### PR TITLE
Add IoU metric, train and validation for one epoch

### PR DESCRIPTION
**Pull request by martinperezmaneiro**

### CODE REVIEW

File: next_sparseconvnet/utils/train_utils.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+
+def train_one_epoch(epoch_id, net, criterion, optimizer, loader):
+    """
+        Trains the net for all the train data one time
+    """
+    net.train()
+    loss_epoch, iou_epoch = [], []
+    for batchid, (coord, ener, label, event) in enumerate(loader):
+        label = label.type(torch.LongTensor) #quitar esto una vez se corrija en el collate
```
Comment (mmkekic) [inline next_sparseconvnet/utils/train_utils.py]: maybe you can change the loaders in this same PR? :)

File: next_sparseconvnet/utils/train_utils.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        optimizer.zero_grad()
+
+        output = net.forward((coord, ener, batch_size))
+
+        loss = criterion(output, label)
+        loss.backward()
+
+        optimizer.step()
+
+        loss_epoch.append(loss.item())
```
Comment (mmkekic) [inline next_sparseconvnet/utils/train_utils.py]: I dont think we want to keep all losses, we only need moving average. So basically you keep summing loss and iou, and divide with number of steps (ie batchid+1) before returning. For iou you can simply keep track of accuracy-per-class for now (ie keep it as vector of size 3).

File: next_sparseconvnet/utils/train_utils.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+        #IoU
+        softmax = torch.nn.Softmax(dim = 1)
+        prediction = torch.argmax(softmax(output), 1)
+        iou_epoch.append(IoU(label.cpu(), prediction.cpu()))
+
+        if batchid%2==0:
```
Comment (mmkekic) [inline next_sparseconvnet/utils/train_utils.py]: The 2 for sure should not be hardcoded. In most cases we just want to know the values at the end of the epoch, but if you prefer you can do something like adding new parameter `print_every = None`, and if is defined then print result every `print_every steps`, other simply print at the end of the epoch?

File: next_sparseconvnet/utils/train_utils.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            progress = f"Train Epoch: {epoch_id} [{batchid*batch_size:5}/{len(loader.dataset)}" +\
+                f" ({int(100*batchid/len(loader)):2}%)]"
+            loss_ = f"\t Loss: {loss.item():.6f}"
+            print(progress + loss_)
+
+        return loss_epoch, iou_epoch
+
+
+def valid_one_epoch(net, loader):
```
Comment (mmkekic) [inline next_sparseconvnet/utils/train_utils.py]: the same comments as for train about loss, iou and printing

File: next_sparseconvnet/utils/train_utils.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    for i in range(len(true)):
+        confusion_matrix[true[i]][pred[i]] += 1
+
+    IoU = []
+    for i in range(nclass):
+        IoU.append((confusion_matrix[i, i] + eps) / (sum(confusion_matrix[:, i]) + sum(confusion_matrix[i, :]) - confusion_matrix[i, i] + eps))
+    return IoU
+
+
+def train_one_epoch(epoch_id, net, criterion, optimizer, loader):
```
Comment (mmkekic) [inline next_sparseconvnet/utils/train_utils.py]: Since you are explicitly calling iuo metrics inside you should rename this to `train_one_epoch_segmentation`or so. Eventually we can merge classification and segmentation in one function, but lets leave that for later

**Review by mmkekic — COMMENTED** [review]
Would be good to add test that this runs without Errors
